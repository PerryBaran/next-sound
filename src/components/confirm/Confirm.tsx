import css from "./confirm.module.css"

interface Props {
  callback: () => void
  setConfirm: (string: string) => void
}

export default function Confirm({ callback, setConfirm }: Props) {
  const handleNo = () => {
    setConfirm("")
  }

  const handleYes = () => {
    setConfirm("")
    callback()
  }

  return (
    <div className={css["container"]}>
      <div className={css["confirm"]}>
        <h3 className={css["are-you-sure"]}>Are You Sure?</h3>
        <div>
          <button type="button" onClick={handleYes} className={css["yes"]}>
            Yes
          </button>
          <button type="button" onClick={handleNo} className={css["no"]}>
            No
          </button>
        </div>
      </div>
    </div>
  )
}
