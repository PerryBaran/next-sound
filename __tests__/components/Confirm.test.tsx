import { fireEvent, render, screen } from "@testing-library/react"
import Confirm from "../../src/components/confirm/Confirm"

describe("Confirm", () => {
  let props = {
    callback: jest.fn(),
    setConfirm: jest.fn()
  }

  beforeEach(() => {
    props = {
      callback: jest.fn(),
      setConfirm: jest.fn()
    }
  })

  test("renders correctly", () => {
    render(<Confirm {...props} />)

    expect(screen.getByRole("heading")).toHaveTextContent(/are you sure?/i)
    expect(screen.getByText(/yes/i)).toBeInstanceOf(HTMLButtonElement)
    expect(screen.getByText(/no/i)).toBeInstanceOf(HTMLButtonElement)
  })

  test("yes button click", () => {
    render(<Confirm {...props} />)

    expect(props.callback).toBeCalledTimes(0)
    expect(props.setConfirm).toBeCalledTimes(0)

    fireEvent.click(screen.getByText(/yes/i))

    expect(props.callback).toBeCalledTimes(1)
    expect(props.setConfirm).toBeCalledTimes(1)
    expect(props.setConfirm).toBeCalledWith("")
  })

  test("no button click", () => {
    render(<Confirm {...props} />)

    expect(props.callback).toBeCalledTimes(0)
    expect(props.setConfirm).toBeCalledTimes(0)

    fireEvent.click(screen.getByText(/no/i))

    expect(props.callback).toBeCalledTimes(0)
    expect(props.setConfirm).toBeCalledTimes(1)
    expect(props.setConfirm).toBeCalledWith("")
  })

  test("snapshot", () => {
    const { asFragment } = render(<Confirm {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
