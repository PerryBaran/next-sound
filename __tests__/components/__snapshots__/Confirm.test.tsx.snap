// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="confirm"
    >
      <h3
        class="are-you-sure"
      >
        Are You Sure?
      </h3>
      <div>
        <button
          class="yes"
          type="button"
        >
          Yes
        </button>
        <button
          class="no"
          type="button"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
